python-backend/app/api/documents/expected/resources/expected_document_uploads.py

```diff
@@ -23,110 +24,126 @@
+        data = {'folder': folder, 'pretty_folder': pretty_folder,
+                'filename': metadata.get('filename')}
+        document_manager_URL = f'{current_app.config["DOCUMENT_MANAGER_URL"]}/document-manager'
+
+        resp = requests.post(
+            url=document_manager_URL,
+            headers={key: value for (key, value)
+                     in request.headers if key != 'Host'},
+            data=data,
+            cookies=request.cookies,
+        )
+
+        response = Response(resp.content, resp.status_code,
+                            resp.raw.headers.items())
+        return response
+     
 
-        self.parser.add_argument('file', type=FileStorage, location='files', action='append')
-        self.parser.add_argument('mine_document_guid', type=str)
-        try:
-            data = self.parser.parse_args()
-
-        except exceptions.RequestEntityTooLarge:
-            return self.create_error_payload(
-                413,
-                f'The maximum file upload size is {current_app.config["MAX_CONTENT_LENGTH"]/1024/1024}MB please ensure all files are this size.'
-            ), 413
-
+    @requires_any_of([MINE_CREATE, MINESPACE_PROPONENT])
+    def put(self, expected_document_guid):
+        if not expected_document_guid:
+            return self.create_error_payload(400, 'Expected Document GUID is required'), 400       
         expected_document = MineExpectedDocument.find_by_exp_document_guid(expected_document_guid)
         if not expected_document:
-            return self.create_error_payload(400, f'expected document not found'), 400
-        mine = Mine.find_by_mine_guid(str(expected_document.mine_guid))
-        document_category = expected_document.required_document.req_document_category.req_document_category
+            return self.create_error_payload(404, 'Expected Document not found'), 404
 
+        data = self.parser.parse_args()
         if data.get('mine_document_guid'):
-            existing_mine_doc = MineDocument.find_by_mine_document_guid(
+            # Associating existing mine document
+            mine_doc = MineDocument.find_by_mine_document_guid(
                 data.get('mine_document_guid'))
-            if not existing_mine_doc:
-                return self.create_error_payload(400, 'mine_document not found'), 400
+            if not mine_doc:
+                return self.create_error_payload(404, 'Mine Document not found'), 404
 
-            expected_document.mine_documents.append(existing_mine_doc)
+            expected_document.mine_documents.append(mine_doc)
+            db.session.commit()
+        elif data.get('document_manager_guid'):
+            # Register and associate a new file upload
+            filename = data.get('filename')
+            if not filename:
+                return self.create_error_payload(400, 'Must supply filename for new file upload'), 400
+            
+            mine_doc = MineDocument(
```

Comment:
Is there no `create_mine_document` method yet?

Reply:
No, and I ddin't think it was worth adding within the scope of this PR.